classes/helper.php

```diff
@@ -79,4 +79,31 @@ public static function enable_feature_policy() {
             @header('Feature-Policy: ' . $featureheader);
         }
     }
+
+    public static function get_violations_listener() : string {
```

Comment:
Is there a good reason these two functions below are not combined? As far as I can tell the first one should have the same conditional logic as the second. Is there ever a reason to record localCspViolationEvents if the popup never happens?

Reply:
You are right. It makes no sense to split those functions up.